shared/src/test/scala/scala/xml/XMLTest.scala

```diff
@@ -369,6 +369,14 @@ Ours is the portal of hope, come as you are."
     </wsdl:definitions>""", wsdlTemplate4("service4", () => "target4") toString)
   }
 
+  @UnitTest
+  def t547: Unit = {
+    // ambiguous toString problem from #547
+
+    val atom: scala.xml.Atom[Unit] = new scala.xml.Atom(())
+    assertTrue("()" == atom.toString || "undefined" == atom.toString)
```

Comment:
You can make the test more precise as follows:
```scala
assertEquals(().toString, atom.toString)
```

Reply:
Point taken. Thanks!